tools/gen_selectedops.py

```diff
@@ -1,80 +1,99 @@
 }
 
 
-def gen_cmake_oplist(opconfig_file, oplist_cmake_file='_selectedoplist.cmake'):
+def gen_cmake_oplist(opconfig_file, oplist_cmake_file):
     ext_domain = "ai.onnx.contrib"  # default_opset_domain()
     new_ext_domain = "com.microsoft.extensions"
     ext_domain_cnt = 0
     cmake_options = set()
-    with open(oplist_cmake_file, 'w') as f:
+    with open(oplist_cmake_file, "w") as f:
         print("# Auto-Generated File, please do not edit!!!", file=f)
-        with open(opconfig_file, 'r') as opfile:
+        with open(opconfig_file, "r") as opfile:
             for _ln in opfile:
                 if _ln.startswith(ext_domain) or _ln.startswith(new_ext_domain):
                     ext_domain_cnt += 1
-                    items = _ln.strip().split(';')
+                    items = _ln.strip().split(";")
                     if len(items) < 3:
                         raise RuntimeError("The malformed operator config file.")
-                    for _op in items[2].split(','):
+                    for _op in items[2].split(","):
                         if not _op:
                             continue  # is None or ""
                         if _op not in OPMAP_TO_CMAKE_FLAGS:
-                            raise RuntimeError("Cannot find the custom operator({})\'s build flags, please update "
-                                               "the OPMAP_TO_CMAKE_FLAGS dictionary.".format(_op))
+                            raise RuntimeError(
+                                "Cannot find the custom operator({})'s build flags, please update "
+                                "the OPMAP_TO_CMAKE_FLAGS dictionary.".format(_op)
+                            )
                         if OPMAP_TO_CMAKE_FLAGS[_op] not in cmake_options:
                             cmake_options.add(OPMAP_TO_CMAKE_FLAGS[_op])
-                            print("set({} ON CACHE INTERNAL \"\")".format(OPMAP_TO_CMAKE_FLAGS[_op]), file=f)
+                            print('set({} ON CACHE INTERNAL "")'.format(OPMAP_TO_CMAKE_FLAGS[_op]), file=f)
         print("# End of Building the Operator CMake variables", file=f)
 
     if ext_domain_cnt == 0:
-        print('[onnxruntime-extensions] warning: lines starting with extension domains of ai.onnx.contrib or '
-              'com.microsoft.extensions in operators config file is 0')
+        print(
+            "[onnxruntime-extensions] warning: lines starting with extension domains of ai.onnx.contrib or "
+            "com.microsoft.extensions in operators config file is 0"
+        )
+
+    print("[onnxruntime-extensions] The cmake tool file has been generated successfully.")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Generate the cmake/_selectedoplist.cmake file from the selected op config file.",
+    )
 
-    print('[onnxruntime-extensions] The cmake tool file has been generated successfully.')
+    parser.add_argument(
+        "selected_op_config_file",
```

Comment:
It's kind of being intentional to always to have the same file to cmake. Given there is opconfig file, we don't add one more file to confuse the user.

Reply:
the script still takes one op config file argument, as it did before. now there is more usage information

Reply:
Given the path is hardcoded in CMakeLists.txt is there any reason it should even be configurable?

Reply:
good point. hardcoded it instead of using a function parameter because it's always the same now. hope that is clearer.

Reply:
it looks like we are all on the same page that there is only one file of cmake op option with a fixed name.